numpy/lib/tests/test_function_base.py

```diff
@@ -1447,6 +1448,38 @@ def test_size_zero_output(self):
         with assert_raises_regex(ValueError, 'new output dimensions'):
             f(x)
 
+    @pytest.mark.skipif(not HAS_REFCOUNT, reason="Python lacks refcounts")
+    def test_leaks(self):
+        # gh-11867
+        # np.frompyfunc(<class method>) creates a reference cycle
+        # that requires a gc collection cycle to break (on CPython 3)
+        import gc
+        gc.disable()
+        try:
+            class A(object):
+                def class_method(self, *args):
+                    return 0
+
+                # Using self.class_method causes reference_cycles
+                def __init__(self):
+                    self.f_vec_otypes = np.vectorize(self.class_method, otypes='d')
+                    self.f_vec = np.vectorize(self.class_method)
+
+            refcount = sys.getrefcount(A.class_method)
+            for i in range(20):
+                a = A()
+                out = a.f_vec_otypes(0)
+                out = a.f_vec(0)
+            a = None
+            if PY2:
+                assert_equal(sys.getrefcount(A.class_method), refcount)
+            else:
+                assert_equal(sys.getrefcount(A.class_method), refcount + 20 * 2)
```

Comment:
Can you extract this 20 and the one above to a variable `N`?

Reply:
done